src/libstore/nar-info-disk-cache.cc

```diff
@@ -186,8 +189,11 @@ class NarInfoDiskCacheImpl : public NarInfoDiskCache
                 auto queryCache(state->queryCache.use()(uri));
                 if (!queryCache.next())
                     return std::nullopt;
+                if (queryCache.getInt(1) + cacheInfoTtl < time(0))
+                    return std::nullopt;
```

Comment:
Just for the sake of consistency, can you implement this check like it's done for the others (with the condition right in the SQL query)?